docs/topics/data-analysis/data-analysis-work-with-data-sources.md

```diff
@@ -0,0 +1,187 @@
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+## Retrieve data from a file
+
+To retrieve data from a file in Kotlin Notebook:
+
+1. Open your Kotlin Notebook file (`.ipynb`).
+2. Import the Kotlin DataFrame library by adding `%use dataframe` in a code cell at the start of your Notebook.
+> Make sure to run the code cell with the `%use dataframe` line before you run any other code cells that rely on the Kotlin DataFrame library.
+>
+{type="note"}
+
+3. Use the `.read()` function from the Kotlin DataFrame library to retrieve data. For example, to read a CSV file, 
+use: `DataFrame.read("example.csv")`.
+
+The `.read()` function automatically detects the input format based on the file extension and content.
+You can also add other arguments to customize the function, such as specifying the delimiter with `delimiter = ';'`.
+
+> For a comprehensive overview of additional file formats and a variety of read functions, see the 
+> [Kotlin DataFrame library documentation](https://kotlin.github.io/dataframe/read.html).
+> 
+{type="tip"}
+
+## Display data
+
+Once you [have the data in your notebook](#retrieve-data-from-a-file), you can easily store it in a variable and access it by running the 
+following in a code cell: 
+
+```kotlin
+val dfJson = DataFrame.read("jsonFile.json")
+dfJson
+```
+
+This code displays the data from the file of your choice, such as CSV, JSON, XLS, XLSX, or Apache Arrow.
+![Display data](display-data.png){width=700}
+
+To gain insights into the structure or schema of your data, apply the `.schema()` function on your DataFrame variable. 
+For example, `dfJson.schema()` lists the type of each column in your JSON dataset.
+
+![Schema example](schema-data-analysis.png){width=700}
+
+You can also use the autocompletion feature in Kotlin Notebook to quickly access and manipulate the properties of your 
+DataFrame. After loading your data, simply type the DataFrame variable followed by a dot to see a list of available columns 
+and their types.
+
+![Available properties](auto-completion-data-analysis.png){width=700}
+
+## Refine data
+
+Among the various operations available in the Kotlin DataFrame library for refining your dataset, key examples include [grouping](https://kotlin.github.io/dataframe/group.html),
+[filtering](https://kotlin.github.io/dataframe/filter.html), [updating](https://kotlin.github.io/dataframe/update.html), 
+and [adding new columns](https://kotlin.github.io/dataframe/add.html). These functions are essential for data analysis, allowing you to organize, clean, and 
+transform your data effectively. 
+
+Let's look at an example where our data includes movie titles and their corresponding release year in the same cell. 
+The goal is to refine this dataset for easier analysis:
+
+1. Load your data into the notebook using the `.read()` function. This example involves reading data from a CSV file named 
```

Comment:
Here and in the list in the last section, code blocks are not aligned with list items